### ISSUE TITLE
Default hierarchy template is prevented from being applied

### ISSUE THREAD

**Issue by jeffdgr8** [rickclephas/KMP-NativeCoroutines#143]
[Starting with Kotlin 1.9.20-RC](https://kotlinlang.org/docs/whatsnew-eap.html#template-for-configuring-multiplatform-projects), the default hierarchy template is applied to KMP projects by default. But the KMP-NativeCoroutines Gradle plugin seems to be preventing this:

```
w: The Default Kotlin Hierarchy Template was not applied to 'project ':shared'':
Explicit .dependsOn() edges were configured for the following source sets:
[iosArm64Main, iosSimulatorArm64Main, iosX64Main, macosArm64Main, macosX64Main]

Consider removing dependsOn-calls or disabling the default template by adding
    'kotlin.mpp.applyDefaultHierarchyTemplate=false'
to your gradle.properties

Learn more about hierarchy templates: https://kotl.in/hierarchy-template
```

No explicit `.dependsOn()` edges are defined in the project and after removing the KMP-NativeCoroutines Gradle plugin, this warning goes away.

**Comment by rickclephas** [comment]
Hi, thanks for opening this issue. This is actually a KSP issue: https://github.com/google/ksp/issues/1569

You can workaround this by explicitly applying the hierarchy template:
```
kotlin {
    applyDefaultHierarchyTemplate()
}
```

**Comment by jeffdgr8** [comment]
Thanks for linking to the root issue. That is the workaround [I employed](https://github.com/jeffdgr8/kotbase/blob/715c67c407aaaa7e967f150183d1ea78e172724f/examples/getting-started/shared/build.gradle.kts#L13) as well.

Seems the fix is to use option 2 suggested [here](https://github.com/google/ksp/issues/1529#issuecomment-1729324887), adding another `srcDir`, rather than the `dependsOn` (option 1) that was implemented in https://github.com/google/ksp/pull/1558.

**Comment by rickclephas** [comment]
This has been fixed in KSP `1.9.20-RC2-1.0.13` and therefore also in [v1.0.0-ALPHA-18-kotlin-1.9.20-RC2](https://github.com/rickclephas/KMP-NativeCoroutines/releases/tag/v1.0.0-ALPHA-18-kotlin-1.9.20-RC2).
